device_calendar/lib/src/device_calendar.dart

```diff
@@ -168,9 +180,11 @@ class DeviceCalendarPlugin {
     return res;
   }
 
+  // TODO: Change this to return a new Result based on the parsed exception
```

Comment:
Handle scenario when the exception is null - create a generic message about 'unknown' platform error